src/mpid/ch4/subconfigure.m4

```diff
@@ -46,8 +46,28 @@ MPID_MAX_ERROR_STRING=512
 
 # $device_args - contains the netmods
 if test -z "${device_args}" ; then
-    AC_MSG_ERROR([Netmod configuration not specified. To build ch4, you must select a netmod:
-    --with-device=ch4:ofi or --with-device=ch4:ucx])
+  dnl default linux builds must choose a netmod.
+  AC_MSG_ERROR([no ch4 netmod selected
+
+  The default ch4 device requires users to choose a netmod. Supported
+  options are ofi (libfabric) and ucx:
+
+    --with-device=ch4:ofi or --with-device=ch4:ucx
+
+  Configure will attempt to discover external libfabric or ucx
```

Comment:
Here we already gave up so saying we "will attempt to ..." is kinda moot :) I guess what we mean is we *failed* at discovering external libfabric or ucx libraries, right?

Reply:
True. One narrow case where it is kind of true is when we find both libfabric and ucx libraries on the system. If the user picks one, then we will still prefer the system installation unless otherwise directed. I guess this is where some additional info would be useful in the error case. I.e. let the user know that we failed to pick a netmod because both libraries were found. I can probably add a patch to be more descriptive.

Reply:
> If the user picks one, then we will still prefer the system installation unless otherwise directed.

If the user picks one, I think we should always try to configure with users choice. We can abort or give warnings in the case it has a conflict system library.

Reply:
What I meant was if the user selects `--with-device=ch4:ofi` then configure will still look for `libfabric` in the system paths before building the embedded version. We won't override the user selection.

Reply:
I modified the text a bit to try and clear this up.